### PR TITLE
Fix usage of deprecated `ruamel.yaml` API

### LINKED ISSUE #921: `ruamel.yaml` vs. `PyYAML`?
@rhfogh wrote:

> One question: There is also an import yaml in HardwareObjects.StateMachine. Should that also be modified?

-- https://github.com/mxcube/mxcubecore/pull/920#pullrequestreview-2009186086

In my environment, `import yaml` imports the ***`yaml` import package*** installed by the ***[`PyYaml`](https://pypi.org/project/PyYAML/) distribution package***. But `PyYAML` is not declared as direct dependency in `pyproject.toml`, which is a bad thing. Obviously, all direct dependencies (the things we import directly) should all be declared in `pyproject.toml`.

In my environment, it happens that `PyYAML` is installed as a dependency of `pre-commit` (and also Sphinx-related things, but that is relatively new, while `pre-commit` has been here longer). So things work fine, but...

1. If we depend on `PyYAML`, we should declare it so in `pyproject.toml`.
2. Maybe we do not need to depend on both Yaml libraries: `ruamel.yaml` and `PyYAML`.

### PR DESCRIPTION
**Pull request by fabcor-maxiv**
Also formatted with `black`.

### CODE REVIEW

**Comment by fabcor-maxiv** [comment]
I am navigating a bit blindly here. I have no idea if this part of the code still runs, not sure how I would trigger it. Since it did not fail before, for me there is no difference. @rhfogh, the impacted part of the code seems to be related to Gphl workflows so you might have better visibility here.

There is another usage of `ruamel.yaml` in `mxcubecore/HardwareRepository.py` but it seems to already conform to the new API.

File: mxcubecore/model/queue_model_objects.py [hunk 1]
```diff
@@ -1609,7 +1611,7 @@ def get_archive_directory(self):
             logging.getLogger("HWR").debug(
                 "PathTemplate (DESY) - (to be defined) directory is %s" % self.directory
             )
-            #archive_directory = self.directory
+            # archive_directory = self.directory
```
Comment (elmjag) [inline mxcubecore/model/queue_model_objects.py]: How about just removing this line :)

Comment (rhfogh) [reply mxcubecore/model/queue_model_objects.py]: Please do.

File: mxcubecore/model/queue_model_objects.py [hunk 2]
```diff
@@ -31,9 +31,11 @@
 
 try:
     from mxcubecore.model import crystal_symmetry
-    import ruamel.yaml as yaml
+    from ruamel.yaml import YAML
```
Comment (rhfogh) [inline mxcubecore/model/queue_model_objects.py:34]: This line should actually be deleted. For another time, how is it you cam maker changes and apply them to the PR?

**Review by rhfogh — APPROVED** [review]
Excellent - and thanks for black-checking this for me.

One question: There is also an 
import yaml
in HardwareObjects.StateMachine. Should that also be modified?

**Comment by fabcor-maxiv** [comment]
> One question: There is also an import yaml in HardwareObjects.StateMachine. Should that also be modified?

Oh! Wow! ... That is another weird one... New ticket, to discuss this: <https://github.com/mxcube/mxcubecore/issues/921>

File: mxcubecore/model/queue_model_objects.py (hunk 1 above)
Comment (fabcor-maxiv) [reply mxcubecore/model/queue_model_objects.py]: Done

**Pull request merged**